lib/models/Location.dart

```diff
@@ -1,39 +1,36 @@
-import 'dart:convert';
 import 'dart:core';
-import 'package:flutter/material.dart';
-import 'package:http/http.dart' as http;
-import '../utils/app_config.dart';
 
 //Model for a location.
 class Location {
   //The id of a location
   final String id;
+
   //The name of a location
   final String name;
+
   //The address of a location
   final String address;
 
-  Location({@required this.id, this.name, this.address});
+  //The information about a location
+  final String info;
+
+  //The tag of a location
+  final String tag;
+
+  Location({this.id, this.name, this.address, this.info, this.tag});
 
   //Creates a location from JSON representation.
   factory Location.fromJson(Map<String, dynamic> json) {
     return Location(
-        id: json['id'], name: json['name'], address: json['address']);
+        id: json['id'],
+        name: json['name'],
+        address: json['address'],
+        info: json['info'],
+        tag: json['tag']);
   }
-}
 
-//Fetches all the locations from the backend as a list by using the baseUrl of [config] and id from [authProvider].
-Future<List<Location>> fetchLocations(BuildContext context) async {
-  final response = await http.get(AppConfig.of(context).baseUrl + '/locations');
-  if (response.statusCode == 200) {
-    String responseBody = response.body;
-    var data = jsonDecode(responseBody)['data'];
-    List<Location> locations =
-        data.map<Location>((e) => Location.fromJson(e)).toList();
-    return locations;
-  } else {
-    throw Exception('Failed to fetch locations.');
-  }
+  Map<String, dynamic> toJson() =>
+      {'id': id, 'name': name, 'address': address, 'info': info, 'tag': tag};
 }
 
 //Converts a list of locations [locations] to a Map containing location ids (key) to locations (value).
```

Comment:
I know this wasn't changed here, but the locationsById function should probably be moved into LocationsProvider?